junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/EnabledIfCondition.java

```
@@ -0,0 +1,195 @@
+		}
+
+		// Bind context-aware names to their current values
+		Accessor configurationParameterAccessor = Accessor.of(context::getConfigurationParameter);
+		Consumer<Bindings> contextBinder = bindings -> {
+			bindings.put(EnabledIf.Bind.JUPITER_TAGS, context.getTags());
+			bindings.put(EnabledIf.Bind.JUPITER_UNIQUE_ID, context.getUniqueId());
+			bindings.put(EnabledIf.Bind.JUPITER_DISPLAY_NAME, context.getDisplayName());
+			bindings.put(EnabledIf.Bind.JUPITER_CONFIGURATION_PARAMETER, configurationParameterAccessor);
+		};
+
+		return evaluate(optionalAnnotation.get(), contextBinder);
+	}
+
+	ConditionEvaluationResult evaluate(EnabledIf annotation, Consumer<Bindings> binder) {
+		Preconditions.notNull(annotation, "annotation must not be null");
+		Preconditions.notNull(binder, "binder must not be null");
+		Preconditions.notEmpty(annotation.value(), "String[] returned by @EnabledIf.value() must not be empty");
+
+		// Find script engine
+		ScriptEngine scriptEngine = findScriptEngine(annotation.engine());
+		logger.debug(() -> "ScriptEngine: " + scriptEngine);
+
+		// Prepare bindings
+		Bindings bindings = scriptEngine.getBindings(ScriptContext.ENGINE_SCOPE);
+		bindings.put(EnabledIf.Bind.SYSTEM_PROPERTY, BIND_SYSTEM_PROPERTY_ACCESSOR);
+		bindings.put(EnabledIf.Bind.SYSTEM_ENVIRONMENT, BIND_SYSTEM_ENVIRONMENT_ACCESSOR);
+		binder.accept(bindings);
+		logger.debug(() -> "Bindings: " + bindings);
+
+		// Build actual script text from annotation properties
+		String script = createScript(annotation, scriptEngine.getFactory().getLanguageName());
+		logger.debug(() -> "Script: " + script);
+
+		return evaluate(annotation, scriptEngine, script);
+	}
+
+	private ConditionEvaluationResult evaluate(EnabledIf annotation, ScriptEngine scriptEngine, String script) {
+		Object result;
+		try {
+			result = scriptEngine.eval(script);
+		}
+		catch (ScriptException e) {
+			logger.warn(e, () -> "Evaluation of @EnabledIf script failed, disabling execution");
```

Shouldn't we let the test fail when evaluating the script fails?

---

Good point. Throw an `JUnitException`?

---

Yep! 👍